src/Settings.cpp

```diff
@@ -155,3 +176,23 @@ bool Settings::getStartInTray() const
 
     return enable;
 }
+
+void Settings::setAutoStart()
+{
+    auto dfPath = getDesktopFilePath();
+    if(dfPath.empty())
+    {
+        std::cerr << "setAutoStart() failed. Couldn't find .desktop file" << std::endl;
+        return;
+    }
+
+    auto homeDir = std::getenv("HOME");
+    auto gfileDestPath = strcat(homeDir, "/.config/autostart/whatsapp-for-linux.desktop");
```

Comment:
Use the `CONFIG_DIR` variable in the free namespace above to get the `config` directory.

Reply:
`CONFIG_DIR` returns `/home/{username}/.config/whatsapp-for-linux`. What i need is `/home/{username}/.config/autostart/whatsapp-for-linux.desktop`. I need to remove `/whatsapp-for-linux` from CONFIG_DIR, which is almost same as what i did.

Since `g_get_user_config_dir()` returns config directory of user and we use that directory in different parts of code we can change `CONFIG_DIR` to `std::string{g_get_user_config_dir()}` only and can make a new variable for `std::string{g_get_user_config_dir()} + "/whatsapp-for-linux";` What do you think about it?

Reply:
Then, we can make `CONFIG_DIR` to return the `.config` dir and have a separate variable for `CONFIG_APP_DIR`. Then you can use that variable instead of calling `getenv`.